docs/howto/object-storage/s3/object-lock.md

```diff
@@ -0,0 +1,361 @@
+but still allow authorized users with special permissions to modify
+the retention settings or delete objects as needed.
+
+To override or remove governance-mode retention settings, users must
+have the `s3:BypassGovernanceRetention` permission. With this permission,
+they can modify or delete objects, regardless of the retention period or
+governance settings that have been established.
+
+#### Compliance mode
+
+> Provides the highest level of protection,
+> and is the best option when storing compliant data.
+
+Compliance mode is recommended when storing compliant data,
+as it prevents objects from being overwritten or deleted
+by any user. If you configure an object with this mode, you
+cannot shorten or change its retention period, ensuring that
+the data remains **secure** and **compliant** with regulatory
+requirements.
+
+> Should be used when you want to ensure that no user, including
+> the administrator user, can delete objects during a defined
+> retention period.
+
+### Legal hold
+
+Legal hold is a feature designed for situations when you are uncertain
+about the length of time you need to retain an object. It can be
+enabled/disabled for any object in a locked bucket, regardless of the
+lock configuration, object retention, or age.
+
+This feature provides the same level of protection as a retention period
+but with no fixed expiration date. Instead, a legal hold will remain in
+effect until you remove it explicitly.
+
+## Prerequisites
+
+- [Aws cli installed](index.md)
+- [S3-compatible credentials](credentials.md)
+
+## Enabling Object Lock
+
+To use the Object Lock feature, you must first create a new bucket.
+
+> Existing buckets cannot have Object Lock enabled.
+
+=== "aws"
+    ```bash
+    aws --profile <region> \
+      --endpoint-url=https://s3-<region>.citycloud.com:8080 \
+      s3api create-bucket \
+      --bucket <bucket-name> \
+      --object-lock-enabled-for-bucket
+    ```
+=== "mc"
+    ```bash
+    mc mb <region>/<bucket> \
+    --with-lock
+    ```
+=== "s3cmd"
+    ```bash
+    s3cmd mb s3://<bucket-name> \
+    --region=<region> \
+    --object-lock
+    ```
+
+> Note that this will enable [bucket versioning](versioning.md) as well.
+
+## Configure the default Object Lock mode and retention period for a bucket
+
+To configure your bucket to either use the **Governance** or
+**Compliance** mode, use one of the following commands:
+
+### Using Governance mode
+
+=== "aws"
+    ```bash
+    aws --profile <region> \
+        --endpoint-url=https://s3-<region>.citycloud.com:8080 \
+        s3api put-object-retention \
+        --bucket <bucket-name> \
+        --key <object-name> \
+        --retention \
+        '{ "Mode": "GOVERNANCE", "RetainUntilDate": "2023-12-31T00:00:00Z" }'
+    ```
+    > For the value of the `RetainUntilDate` parameter, use the
+    > [ISO 8601 date-time representation](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations)
+    > (`Y-m-dTH:M:S.%3fZ`) format.
```

Comment:
I think the `Y-m-dTH:M:S.%3fZ` format string is more confusing than beneficial. :) Consider dropping it.